File: march-2026/3-12-world-wide-web/world_wide_web.js
// Word Wide Web 🌐
// Codédex

function checkUrl(address) {
  if (!(address.startsWith("https://") || address.startsWith("http://"))) {
    return "invalid";
  };
  
  let domain = address.split("://");
  domain = domain[1];
  if (!domain.includes(".")) {
    return "invalid";
  };
  
  const allowedChars = /^[a-zA-Z0-9-.]+$/.test(domain);
  if (!allowedChars) {
    return "invalid";
  }
  
  return "valid";
};

File: march-2026/3-12-world-wide-web/world-wide-web.py
def check_url(address):
  # check if https or http
  # if so, domain takes the url without the http/s
  if address.startswith("https://"):
    domain = address[8:]
  elif address.startswith("http://"):
    domain = address[7:]
  else:
    return "invalid"
  
  # if not any dot, invalid
  if "." not in domain:
    return "invalid"
  for char in domain:
    # if char is not either a-z, A-Z, 0-9, - or . -> invalid
    if not (char.isalpha() or char.isdigit() or char == "-" or char == "."):
      return "invalid"
    
  return "valid"